com.swookiee.runtime.metrics.prometheus/src/main/java/com/swookiee/runtime/metrics/prometheus/CollectorRegistryInventory.java

```diff
@@ -0,0 +1,122 @@
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.BundleListener;
+import org.osgi.service.component.annotations.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+
+@Component(service = {CollectorRegistryInventory.class})
+public class CollectorRegistryInventory implements BundleListener {
+
+    private static final Logger logger = LoggerFactory.getLogger(CollectorRegistryInventory.class);
+    private static final String BUNDLE_ID = "service.bundleid";
+    private BundleContext bundleContext;
+
+    private Map<String, CollectorRegistry> registeredRegistries = new ConcurrentHashMap<>();
+    private Function<String, CollectorRegistry> registryInstantiator = new Function<String, CollectorRegistry>() {
+        @Override
+        public CollectorRegistry apply(String bundle) {
+            return new CollectorRegistry();
+        }
+    };
+
+    @Activate
+    public void activate(final BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+        this.bundleContext.addBundleListener(this);
+        logger.info("Activated metric collector!");
+    }
+
+    @Deactivate
+    public void deactivate() {
+        for (CollectorRegistry registry : registeredRegistries.values()) {
+            registry.clear();
+        }
+        registeredRegistries.clear();
+        logger.info("Deactivated metric collector!");
+    }
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    public void registerCollection(CollectorCollection collectorCollection, Map<String, Object> properties) {
+        CollectorRegistry registry = getRegistry(properties);
+        for (Collector collector : collectorCollection.getCollectors()) {
+            registry.register(collector);
+        }
+    }
+
+    public void unregisterCollection(CollectorCollection collectorCollection, Map<String, Object> properties) {
+        CollectorRegistry registry = getRegistry(properties);
+        for (Collector collector : collectorCollection.getCollectors()) {
+            registry.unregister(collector);
+        }
+    }
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    public void registerCollector(Collector metricsCollection, Map<String, Object> properties) {
+        getRegistry(properties).register(metricsCollection);
+    }
+
+    public void unregisterCollector(Collector metricsCollection, Map<String, Object> properties) {
+        getRegistry(properties).unregister(metricsCollection);
+    }
+
+    public CollectorRegistry getCollectorRegistry(String bundle) {
+        return registeredRegistries.get(bundle);
+    }
+
+    public List<String> getRegisteredBundles() {
+        return new ArrayList<>(registeredRegistries.keySet());
```

Comment:
Why not using the Set? The using code should be doable without this transformation